src/core/utils/compatibility.js

```diff
@@ -349,34 +354,33 @@ export const correctedLocationForPlatform = ({
 
   const { browser, os } = userAgentInfo;
 
-  let currentClientApp;
-  let newClientApp;
-
-  if (os.name === USER_AGENT_OS_IOS) {
+  if (isFirefoxForIOS(userAgentInfo) || !isFirefox({ userAgentInfo })) {
     return null;
   }
 
   if (
-    os.name === USER_AGENT_OS_ANDROID &&
-    browser.name === USER_AGENT_BROWSER_FIREFOX &&
-    clientApp === CLIENT_APP_FIREFOX
+    // Android browser on desktop site.
+    (isFirefoxForAndroid(userAgentInfo) && clientApp === CLIENT_APP_FIREFOX) ||
+    // Android browser on page other than Home or Search.
+    (isFirefoxForAndroid(userAgentInfo) &&
+      !isHomePage &&
+      !location.pathname.includes('/search/'))
   ) {
-    currentClientApp = CLIENT_APP_FIREFOX;
-    newClientApp = CLIENT_APP_ANDROID;
+    // Redirect to `android` home page.
+    return MOBILE_HOME_PAGE_LINK;
```

Comment:
I think I liked the previous approach before because the locale was kept in the URL if I understand the red code correctly.

Reply:
Fixed with the next commit.